### DISCUSSION TITLE
How is the access token stored on the client, in the tutorial "Simple OAuth2 with Password and Bearer"

### DISCUSSION THREAD

**Discussion by ben519** [fastapi/fastapi#6427] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

from fastapi import Depends, FastAPI, HTTPException, status
from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
from pydantic import BaseModel

fake_users_db = {
    "johndoe": {
        "username": "johndoe",
        "full_name": "John Doe",
        "email": "johndoe@example.com",
        "hashed_password": "fakehashedsecret",
        "disabled": False,
    },
    "alice": {
        "username": "alice",
        "full_name": "Alice Wonderson",
        "email": "alice@example.com",
        "hashed_password": "fakehashedsecret2",
        "disabled": True,
    },
}

app = FastAPI()


def fake_hash_password(password: str):
    return "fakehashed" + password


oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")


class User(BaseModel):
    username: str
    email: Optional[str] = None
    full_name: Optional[str] = None
    disabled: Optional[bool] = None


class UserInDB(User):
    hashed_password: str


def get_user(db, username: str):
    if username in db:
        user_dict = db[username]
        return UserInDB(**user_dict)


def fake_decode_token(token):
    # This doesn't provide any security at all
    # Check the next version
    user = get_user(fake_users_db, token)
    return user


async def get_current_user(token: str = Depends(oauth2_scheme)):
    user = fake_decode_token(token)
    if not user:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Invalid authentication credentials",
            headers={"WWW-Authenticate": "Bearer"},
        )
    return user


async def get_current_active_user(current_user: User = Depends(get_current_user)):
    if current_user.disabled:
        raise HTTPException(status_code=400, detail="Inactive user")
    return current_user


@app.post("/token")
async def login(form_data: OAuth2PasswordRequestForm = Depends()):
    user_dict = fake_users_db.get(form_data.username)
    if not user_dict:
        raise HTTPException(status_code=400, detail="Incorrect username or password")
    user = UserInDB(**user_dict)
    hashed_password = fake_hash_password(form_data.password)
    if not hashed_password == user.hashed_password:
        raise HTTPException(status_code=400, detail="Incorrect username or password")

    return {"access_token": user.username, "token_type": "bearer"}


@app.get("/users/me")
async def read_users_me(current_user: User = Depends(get_current_active_user)):
    return current_user
```

### Description

I'm pretty new to FastAPI and OAuth in general. I just worked through the tutorial ["Simple OAuth2 with Password and Bearer"](https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/) and it mostly made sense, but there was one step that felt like magic to me.. 

**How does the access token get stored onto the client and subsequently get passed into the client's requests?**

My understanding of the flow is that it's basically

1. User authenticates with their username and password (these get POST'ed to the `/token` endpoint).
2. User's credentials are validated, and the `/token` endpoint returns the access token (`johndoe`) inside some JSON. (This is how the user receives his access token)
3. ???
4. User make a subsequent request to a private endpoint, like `GET /users/me`. The user's request includes the header `Authorization: Bearer johndoe`. (I don't think the docs mention this, but it's what I've gathered from inspecting the request in Chrome Developer Tools)
5. The authorization token is then used to lookup the user who made the request in (4)

Step (3) is the part that I don't understand. How does the access token seemingly get stored on the client? Thanks!

### Operating System

macOS

### FastAPI Version

0.70.0

### Python Version

3.9.7

**Comment by GhuulCZ** [comment]
I'm pretty new to FastAPI as well so I might be wrong but I believe that the answer is that it is not stored anywhere. You need to capture the token and add it to Authorization header when you need to use it. You also need to take care of token expiration and request new one when it expires (or before that).

**Comment by ben519** [comment]
> You need to capture the token and add it to Authorization header when you need to use it

If this is true, how does the `/users/me` endpoint work in the [demonstration at the end of the tutorial](https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/#see-it-in-action)? Specifically, what line(s) of code are capturing the token and adding it to the Authorization header?  Thanks

**Comment by GhuulCZ** [comment]
I assume you mean it works when using the authorization form. I have not checked the code but I guess that the access token is stored in some variable (it is basically one page js client application) and kept during the "session". If you reload the page you will loose the authorization...

**Comment by ben519** [comment, marked as answer]
It seems that Swagger persists the token to [local storage](https://stackoverflow.com/a/69853435/2146894).
